src/smt/process_assertions.cpp

```diff
@@ -738,7 +738,8 @@ Node ProcessAssertions::expandDefinitions(
         theory::Theory* t = d_smt.d_theoryEngine->theoryOf(node);
 
         Assert(t != NULL);
-        node = t->expandDefinition(n);
+        TrustNode trn = t->expandDefinition(n);
+        node = trn.isNull() ? Node(n) : trn.getNode();
```

Comment:
shouldn't you be using `static_cast` here (and all other such occurrences in the PR)?

Reply:
No, this is not a cast, it's a constructor.

Reply:
I see. I have done this via casting, any idea which would be better?

Reply:
Casting is more roundabout and is generally worse.

Reply:
OK, good to know.

Reply:
I agree. I'd only use casts for pointers and other primitive types.